### PR TITLE
modules: chroot: Add cgroup binds

### LINKED ISSUE #38: cgroup problem when running docker
After installing docker and running dockerd, the following error happens
```
### RUN sudo dockerd
INFO[2021-11-01T19:58:45.282788937Z] Starting up                                  
INFO[2021-11-01T19:58:45.307226380Z] libcontainerd: started new containerd process  pid=12259
INFO[2021-11-01T19:58:45.307411211Z] parsed scheme: "unix"                         module=grpc
INFO[2021-11-01T19:58:45.307458044Z] scheme "unix" not registered, fallback to default scheme  module=grpc
INFO[2021-11-01T19:58:45.307559820Z] ccResolverWrapper: sending update to cc: {[{unix:///var/run/docker/containerd/containerd.sock  <nil> 0 <nil>}] <nil> <nil>}  module=grpc
INFO[2021-11-01T19:58:45.307622523Z] ClientConn switching balancer to "pick_first"  module=grpc
INFO[2021-11-01T19:58:46.043406825Z] starting containerd                           revision=e25210fe30a0a703442421b0f60afac609f950a3 version=1.4.9
INFO[2021-11-01T19:58:46.165887464Z] loading plugin "io.containerd.content.v1.content"...  type=io.containerd.content.v1
INFO[2021-11-01T19:58:46.166480085Z] loading plugin "io.containerd.snapshotter.v1.aufs"...  type=io.containerd.snapshotter.v1
INFO[2021-11-01T19:58:46.224271936Z] skip loading plugin "io.containerd.snapshotter.v1.aufs"...  error="aufs is not supported (modprobe aufs failed: exit status 1 \"modprobe: FATAL: Module aufs not found in directory /lib/modules/4.19.127-navigator-v7l+\\n\"): skip plugin" type=io.containerd.snapshotter.v1
INFO[2021-11-01T19:58:46.224460785Z] loading plugin "io.containerd.snapshotter.v1.btrfs"...  type=io.containerd.snapshotter.v1
INFO[2021-11-01T19:58:46.225143997Z] skip loading plugin "io.containerd.snapshotter.v1.btrfs"...  error="path /var/lib/docker/containerd/daemon/io.containerd.snapshotter.v1.btrfs (ext4) must be a btrfs filesystem to be used with the btrfs snapshotter: skip plugin" type=io.containerd.snapshotter.v1
INFO[2021-11-01T19:58:46.225255033Z] loading plugin "io.containerd.snapshotter.v1.devmapper"...  type=io.containerd.snapshotter.v1
WARN[2021-11-01T19:58:46.225348865Z] failed to load plugin io.containerd.snapshotter.v1.devmapper  error="devmapper not configured"
INFO[2021-11-01T19:58:46.225430012Z] loading plugin "io.containerd.snapshotter.v1.native"...  type=io.containerd.snapshotter.v1
INFO[2021-11-01T19:58:46.225739026Z] loading plugin "io.containerd.snapshotter.v1.overlayfs"...  type=io.containerd.snapshotter.v1
INFO[2021-11-01T19:58:46.226527459Z] loading plugin "io.containerd.snapshotter.v1.zfs"...  type=io.containerd.snapshotter.v1
INFO[2021-11-01T19:58:46.226969693Z] skip loading plugin "io.containerd.snapshotter.v1.zfs"...  error="path /var/lib/docker/containerd/daemon/io.containerd.snapshotter.v1.zfs must be a zfs filesystem to be used with the zfs snapshotter: skip plugin" type=io.containerd.snapshotter.v1
INFO[2021-11-01T19:58:46.227066821Z] loading plugin "io.containerd.metadata.v1.bolt"...  type=io.containerd.metadata.v1
WARN[2021-11-01T19:58:46.227271467Z] could not use snapshotter devmapper in metadata plugin  error="devmapper not configured"
INFO[2021-11-01T19:58:46.227349077Z] metadata content store policy set             policy=shared
INFO[2021-11-01T19:58:46.270543008Z] loading plugin "io.containerd.differ.v1.walking"...  type=io.containerd.differ.v1
INFO[2021-11-01T19:58:46.270696562Z] loading plugin "io.containerd.gc.v1.scheduler"...  type=io.containerd.gc.v1
INFO[2021-11-01T19:58:46.270854930Z] loading plugin "io.containerd.service.v1.introspection-service"...  type=io.containerd.service.v1
INFO[2021-11-01T19:58:46.271050705Z] loading plugin "io.containerd.service.v1.containers-service"...  type=io.containerd.service.v1
INFO[2021-11-01T19:58:46.271139370Z] loading plugin "io.containerd.service.v1.content-service"...  type=io.containerd.service.v1
INFO[2021-11-01T19:58:46.271250276Z] loading plugin "io.containerd.service.v1.diff-service"...  type=io.containerd.service.v1
INFO[2021-11-01T19:58:46.271334349Z] loading plugin "io.containerd.service.v1.images-service"...  type=io.containerd.service.v1
INFO[2021-11-01T19:58:46.271411737Z] loading plugin "io.containerd.service.v1.leases-service"...  type=io.containerd.service.v1
INFO[2021-11-01T19:58:46.271505569Z] loading plugin "io.containerd.service.v1.namespaces-service"...  type=io.containerd.service.v1
INFO[2021-11-01T19:58:46.271588067Z] loading plugin "io.containerd.service.v1.snapshots-service"...  type=io.containerd.service.v1
INFO[2021-11-01T19:58:46.271661918Z] loading plugin "io.containerd.runtime.v1.linux"...  type=io.containerd.runtime.v1
INFO[2021-11-01T19:58:46.272246039Z] loading plugin "io.containerd.runtime.v2.task"...  type=io.containerd.runtime.v2
INFO[2021-11-01T19:58:46.272667681Z] loading plugin "io.containerd.monitor.v1.cgroups"...  type=io.containerd.monitor.v1
INFO[2021-11-01T19:58:46.273782332Z] loading plugin "io.containerd.service.v1.tasks-service"...  type=io.containerd.service.v1
INFO[2021-11-01T19:58:46.273904849Z] loading plugin "io.containerd.internal.v1.restart"...  type=io.containerd.internal.v1
INFO[2021-11-01T19:58:46.274082791Z] loading plugin "io.containerd.grpc.v1.containers"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.274168771Z] loading plugin "io.containerd.grpc.v1.content"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.274246029Z] loading plugin "io.containerd.grpc.v1.diff"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.274321380Z] loading plugin "io.containerd.grpc.v1.events"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.274397971Z] loading plugin "io.containerd.grpc.v1.healthcheck"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.274476970Z] loading plugin "io.containerd.grpc.v1.images"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.274553228Z] loading plugin "io.containerd.grpc.v1.leases"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.274607487Z] loading plugin "io.containerd.grpc.v1.namespaces"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.274660375Z] loading plugin "io.containerd.internal.v1.opt"...  type=io.containerd.internal.v1
INFO[2021-11-01T19:58:46.275082795Z] loading plugin "io.containerd.grpc.v1.snapshots"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.275197460Z] loading plugin "io.containerd.grpc.v1.tasks"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.275258237Z] loading plugin "io.containerd.grpc.v1.version"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.275317440Z] loading plugin "io.containerd.grpc.v1.introspection"...  type=io.containerd.grpc.v1
INFO[2021-11-01T19:58:46.275951967Z] serving...                                    address=/var/run/docker/containerd/containerd-debug.sock
INFO[2021-11-01T19:58:46.276146965Z] serving...                                    address=/var/run/docker/containerd/containerd.sock.ttrpc
INFO[2021-11-01T19:58:46.276317592Z] serving...                                    address=/var/run/docker/containerd/containerd.sock
INFO[2021-11-01T19:58:46.276454868Z] containerd successfully booted in 0.236700s  
INFO[2021-11-01T19:58:46.297846577Z] parsed scheme: "unix"                         module=grpc
INFO[2021-11-01T19:58:46.297912947Z] scheme "unix" not registered, fallback to default scheme  module=grpc
INFO[2021-11-01T19:58:46.297981705Z] ccResolverWrapper: sending update to cc: {[{unix:///var/run/docker/containerd/containerd.sock  <nil> 0 <nil>}] <nil> <nil>}  module=grpc
INFO[2021-11-01T19:58:46.298023315Z] ClientConn switching balancer to "pick_first"  module=grpc
INFO[2021-11-01T19:58:46.300877293Z] parsed scheme: "unix"                         module=grpc
INFO[2021-11-01T19:58:46.300939162Z] scheme "unix" not registered, fallback to default scheme  module=grpc
INFO[2021-11-01T19:58:46.300999384Z] ccResolverWrapper: sending update to cc: {[{unix:///var/run/docker/containerd/containerd.sock  <nil> 0 <nil>}] <nil> <nil>}  module=grpc
INFO[2021-11-01T19:58:46.301035346Z] ClientConn switching balancer to "pick_first"  module=grpc
INFO[2021-11-01T19:58:46.468130545Z] [graphdriver] using prior storage driver: overlay2 
WARN[2021-11-01T19:58:46.579388772Z] Your kernel does not support cgroup memory limit 
WARN[2021-11-01T19:58:46.579462845Z] Unable to find cpu cgroup in mounts          
WARN[2021-11-01T19:58:46.579807969Z] Unable to find blkio cgroup in mounts        
WARN[2021-11-01T19:58:46.580029522Z] Unable to find cpuset cgroup in mounts       
WARN[2021-11-01T19:58:46.580156038Z] Unable to find pids cgroup in mounts         
INFO[2021-11-01T19:58:46.581864643Z] stopping healthcheck following graceful shutdown  module=libcontainerd
INFO[2021-11-01T19:58:46.582015974Z] stopping event stream following graceful shutdown  error="context canceled" module=libcontainerd namespace=plugins.moby
failed to start daemon: Devices cgroup isn't mounted
```

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #38 

Allow the usage of dockerd on pimod

### CODE REVIEW

File: modules/chroot.sh [hunk 1]
```diff
@@ -13,6 +13,8 @@ chroot_setup() {
   mount --bind /sys "${CHROOT_MOUNT}/sys"
   mount --bind /proc "${CHROOT_MOUNT}/proc"
   mount --bind /dev/pts "${CHROOT_MOUNT}/dev/pts"
+  mount --bind /proc/cgroups "${CHROOT_MOUNT}/proc/cgroups"
```
Comment (oxzi) [inline modules/chroot.sh:16]: This entry shouldn't be necessary, as it should already be covered by the previous bind mount of `/proc`.

File: modules/chroot.sh [hunk 2]
```diff
@@ -13,6 +13,8 @@ chroot_setup() {
...
+  mount --rbind /sys/fs/cgroup "${CHROOT_MOUNT}/sys/fs/cgroup"
```
Comment (oxzi) [inline modules/chroot.sh:17]: The `--rbind` is a good catch for `/sys`. Maybe you want to change the previous bind mount for `/sys` to an `--rbind` mount?

**Review by oxzi — CHANGES_REQUESTED** [review]
Thanks for raising this issue and suggesting this fix. Would you please try the suggested change? Btw, this should hopefully also satisfy the CI.

__EDIT:__ Might you please also add this change/fix to the CHANGELOG? Thanks.

**Comment by patrickelectric** [comment]
@oxzi done!

File: modules/chroot.sh [hunk 3]
```diff
@@ -13,6 +13,7 @@ chroot_setup() {
   mount --bind /sys "${CHROOT_MOUNT}/sys"
   mount --bind /proc "${CHROOT_MOUNT}/proc"
   mount --bind /dev/pts "${CHROOT_MOUNT}/dev/pts"
+  mount --rbind /sys "${CHROOT_MOUNT}/sys"
```
Comment (oxzi) [inline modules/chroot.sh:16]: Thanks a lot for your fast changes.

However, now we are bind mounting `/sys` twice, as it was already bind mounted before in line 13. Would you please remove the other, non `--rbind` mount?

**Review by oxzi — CHANGES_REQUESTED** [review]

**Comment by oxzi** [comment]
I am not quite sure why the CI is unhappy about the umount, as it works perfectly on my machine. I will try to investigate this one later.

**Comment by patrickelectric** [comment]
@oxzi It appears to be working fine on CI, I have just tested on my fork.

**Comment by patrickelectric** [comment]
@oxzi can you check it now ?

**Review by oxzi — APPROVED** [review]
LGTM!

Thanks a lot and please excuse the delay.

**Pull request merged**